.github/compose/ci.docker-compose.yml

```diff
@@ -34,14 +34,6 @@ services:
     <<: *tests-template
     image: registry.gitlab.com/buildstream/buildstream-docker-images/testsuite-debian:10-${CI_IMAGE_VERSION:-latest}
 
-  ubuntu-18.04:
-    <<: *tests-template
-    image: registry.gitlab.com/buildstream/buildstream-docker-images/testsuite-ubuntu:18.04-${CI_IMAGE_VERSION:-latest}
-
```

Comment:
Would not be better to still check bst-1 runs ok in 18.04 with python 3.7? Same for centos-7.7?

Reply:
@jjardon on CentOS 7.7 that would be 3.8 as that is the stack provided, not 3.7. That could in theory be done but requires significant amount of development work in the images and possibly in CI. Is there real need to keep supporting 18.04? Even flatpak-builder dropped support for that.

Reply:
The most important aspect, in my opinion, is that we keep testing Python 3.7 in CI. We can't but also shouldn't have to test all distro/Python combination. I think it's preferable to test distros with their default Python 3 version as that's more likely what people actually use.

Python 3.7 is already covered by Debian 10, which is expected to get LTS support until June 2024, which is after the Python 3.7 EOL date (June 2023). While there is a Python 3.7 package for Ubuntu 18.04, it's not the default version and it's in the 'universe' archive, which means that it's not guaranteed to be supported by Canonical, as far as I can tell. We now have a Ubuntu 20.04 (Python 3.8) Docker image for master, which we could also use for 1.x.

CentOS 7.7 (1908) is no longer supported. We could upgrade to the still supported CentOS 7.9 (2009) and install Python 3.8 from the software collection. However, I'd like to drop CentOS 7 from the Docker images for BuildStream master to lower maintenance cost and image build time. This means that we'd have to maintain a separate branch of bst-docker-images for 1.x. In my opinion, it would make more sense to upgrade to CentOS Stream 8 with Python 3.8 or 3.9, which would be useful for both 1.x and master.

Reply:
> The most important aspect, in my opinion, is that we keep testing Python 3.7 in CI

Yeah agree with that, and as that's covered with Debian 10 I think I'm ok with removing ubuntu-18.04

For CentOS, taking in account current usage of bst-1, I think is still worth to try to keep CentOS 7.x tests running, but I'd like to hear @ssssam opinions here

Reply:
@ssssam mentioned that they will be using CentOS 7 but will anyway upgrade to a more recent version of Python, which means that the current outdated and broken CentOS 7 test is not useful to them either. I suggest we go ahead and merge this now but try to re-add a new CentOS 7 test (with Python 3.8) in a follow-up.

Reply:
wfm!